x-pack/plugin/vectors/src/main/java/org/elasticsearch/xpack/vectors/query/VectorDVIndexFieldData.java

```diff
@@ -26,10 +25,29 @@
 import org.elasticsearch.search.sort.SortOrder;
 
 
-public class VectorDVIndexFieldData extends DocValuesIndexFieldData implements IndexFieldData<VectorDVLeafFieldData> {
+public class VectorDVIndexFieldData implements IndexFieldData<VectorDVLeafFieldData> {
+
+    protected final Index index;
+    protected final String fieldName;
 
     public VectorDVIndexFieldData(Index index, String fieldName) {
-        super(index, fieldName);
+        this.index = index;
+        this.fieldName = fieldName;
+    }
+
+    @Override
+    public final String getFieldName() {
+        return fieldName;
+    }
+
+    @Override
+    public final void clear() {
+        // can't do
```

Comment:
I see what you mean about copying method around and maybe making an abstract base class at some point. But the copying is fine by me as an intermediate step to untwisting all of this.